### DISCUSSION TITLE
Remix CLI node version detection always erroring, says old version

### DISCUSSION THREAD

**Discussion by davek-eql** [remix-run/remix#8593] [Q&A]
When I try to run `npx remix build`, I get an error saying an old incompatible version of Node is detected, v17.0.1.  I have 21.6.1 installed.

I am a Mac OS user who has Node installed with Homebrew and NVM.

I believe that the check it's failing is at https://www.npmjs.com/package/@remix-run/dev?activeTab=code, lines 102-107:

```
async function run(argv = process.argv.slice(2)) {
  // Check the node version
  let versions = process.versions;
  if (versions && versions.node && semver__default["default"].major(versions.node) < 18) {
    throw new Error(`️🚨 Oops, Node v${versions.node} detected. Remix requires a Node version greater than 18.`);
  }
```

I was trying to use Shopify Hydrogen, but when that kept failing switched over to the Remix quickstart where I encountered the same problem:

https://remix.run/docs/en/main/start/quickstart#build-and-run

I added a simple "check-version.js" file to the project whose sole contents are `console.log(process.version);` in order to see what version the process used.  When run from terminal it used v21.6.1.

Here are some traces:

```
remix-starter npx remix build
Error: ️🚨 Oops, Node v17.0.1 detected. Remix requires a Node version greater than 18.
    at Object.run (/Users/<redacted-username>/<redacted-projectname>/headless/selfhost-headless/remix-starter/node_modules/@remix-run/dev/dist/cli/run.js:106:11)
    at Object.<anonymous> (/Users/<redacted-username>/<redacted-projectname>/headless/selfhost-headless/remix-starter/node_modules/@remix-run/dev/dist/cli.js:16:11)
    at Module._compile (node:internal/modules/cjs/loader:1095:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1147:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47
➜  remix-starter node --version
v21.6.1
➜  remix-starter nvm --version
0.39.7
➜  remix-starter npm --version
10.2.4
➜  remix-starter npx --version
10.2.4
➜  remix-starter node check-version.js 
v21.6.1
➜  remix-starter node
Welcome to Node.js v21.6.1.
Type ".help" for more information.
> 
```
And a look at my project structure, straight from the quickstart + the version check file:
```
➜  remix-starter tree -I node_modules
.
├── app
│   └── root.jsx
├── check-version.js
├── package-lock.json
└── package.json

```

**Comment by davek-eql** [comment]
Nvm, looks like at some point I had NVM installed multiple places?  Found a "node-bin-setup" directory that used `installArchSpecificPackage.js`, which was requiring a specific version of Node:  `require('node-bin-setup')("17.0.1", require)`.  Guess on `npx remix build` it was hitting that one somehow while other uses were not.

**Reply by AnthonyMrt** [reply]
hello and finally, what you do?  did you delete the installArchSpecificPackage.js? cause i facing the same issue with remix npm run dev

**Reply by mvangelist** [reply]
I found the file in a node_module folder installed at the root of my drive. I just deleted the folder directory and my project started picking up my node from my Program Files location.

**Reply by tranvanlong6677** [reply]
@mvangelist oh thanks for your suggestion, I deleted node_module folder and run dev, my project ran successfully
